Sources/GRPCHTTP2Core/GRPCMessageFramer.swift

```diff
@@ -69,15 +74,21 @@ struct GRPCMessageFramer {
 
     var requiredCapacity = 0
     for message in self.pendingMessages {
-      requiredCapacity += message.count + Self.metadataLength
+      requiredCapacity += message.bytes.count + Self.metadataLength
     }
     self.writeBuffer.clear(minimumCapacity: requiredCapacity)
 
+    var pendingWritePromise: EventLoopPromise<Void>?
     while let message = self.pendingMessages.pop() {
-      try self.encode(message, compressor: compressor)
+      try self.encode(message.bytes, compressor: compressor)
+      if pendingWritePromise == nil {
+        pendingWritePromise = message.promise
+      } else {
+        pendingWritePromise?.futureResult.cascade(to: message.promise)
+      }
```

Comment:
nit: prefer `if let`:

```swift
if let pendingWritePromise = pendingWritePromise {
  // ...
} else {
  // ...
}
```

Reply:
I don't think I can do this. First of all, I'm overriding the value of `pendingWritePromise`, so it should be `if var ...`.
But also, since I've defined `pendingWritePromise` a couple of lines above this, wouldn't this declaration shadow that one?

Reply:
> I don't think I can do this. First of all, I'm overriding the value of `pendingWritePromise`, so it should be `if var ...`. 

You only set `pendingWritePromise` if it's nil, so that'd be done in the `else` branch.

> But also, since I've defined `pendingWritePromise` a couple of lines above this, wouldn't this declaration shadow that one?

Yeah, but you can use a different name, e.g.:

```swift
if let promise = pendingWritePromise {
  promise.futureResult.cascade(to: message.promise)
} else {
  pendingWritePromise = message.promise
}
```

---

The aim is to remove the unnecessary `?` in the original `else` branch. You can reason that the value must be present (because it's `nil` in the other branch) which means you could safely replace it with `!`, at that point you can just flip the check and use `if let`.

Reply:
Ah I see what you mean. Thanks. I've changed this.